akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -833,6 +876,14 @@ abstract class ShardCoordinator(
         if (ok) {
           log.debug("{}: Shard [{}] deallocation completed successfully.", typeName, shard)
 
+          // ack to external trigger of rebalance/stop
+          waitingForRebalanceAcks.get(shard) match {
```

Comment:
what should we do in the case of `!ok`? Now they are kept in the `waitingForRebalanceAcks`

Reply:
Hmmm, yeah, I was thinking there is no nack back on failure and caller is responsible for retrying on timeout, but we anyway need a timeout to not leak these.

Reply:
I think it's fine to have no reply, just timeout, for the caller. Retries are idempotent. The leak is more concerning and since more requests can be added while in progress we can't just clear the whole `waitingForRebalanceAcks`.

Reply:
Ok. Some form of identifier for the set of waiting actors and put that along the set of waiting and in the timeout message as well will do the trick.